### ISSUE TITLE
DOC: fix PR07 errors in docstrings

### ISSUE THREAD

**Issue by jordan-d-murphy** [pandas-dev/pandas#57420]
Pandas has a script for validating docstrings:

Currently, some methods fail the PR07 check.

https://github.com/pandas-dev/pandas/blob/5fff2cd4818514a2d217e4202e1d78de2290c207/ci/code_checks.sh#L144-L258

The task here is:

- take 2-4 methods
- run: scripts/validate_docstrings.py --format=actions --errors=PR07 `method-name`
- check if validation docstrings passes for those methods, and if it’s necessary fix the docstrings according to whatever error is reported
- remove those methods from code_checks.sh
- commit, push, open pull request

Please don't comment take as multiple people can work on this issue. You also don't need to ask for permission to work on this, just comment on which methods are you going to work.

If you're new contributor, please check the [contributing guide](https://pandas.pydata.org/docs/dev/development/contributing.html)

**Comment by jordan-d-murphy** [comment]
I don't have permission to add them, but this could probably use some labels:

- CI
- Docs
- good first issue

**Comment by jordan-d-murphy** [comment]
Addresses https://github.com/pandas-dev/pandas/pull/57357

**Comment by jordan-d-murphy** [comment]
opened a fix for `pandas.DataFrame.align`

**Comment by jordan-d-murphy** [comment]
opening a fix for 

```
pandas.DataFrame.get
pandas.DataFrame.rolling
pandas.DataFrame.to_hdf
```

**Comment by jordan-d-murphy** [comment]
Hi all, if https://github.com/pandas-dev/pandas/pull/57826 gets merged in, I might be reworking our approach here; this would look like closing the following issues: 

https://github.com/pandas-dev/pandas/issues/57443
https://github.com/pandas-dev/pandas/issues/57438
https://github.com/pandas-dev/pandas/issues/57420
https://github.com/pandas-dev/pandas/issues/57417
https://github.com/pandas-dev/pandas/issues/57416
https://github.com/pandas-dev/pandas/issues/57111

And opening a new issue to address these based on the new approach. 

tl;dr 

the work can still be done, but probably under a new ticket once  #57826 is merged in

**Comment by jordan-d-murphy** [comment]
Closed by:
[CI: speedup docstring check consecutive runs #57826](https://github.com/pandas-dev/pandas/pull/57826)
[CI: Better error control in the validation of docstrings #57879](https://github.com/pandas-dev/pandas/pull/57879) 

Opening a new issue to fix docstring errors.

**Comment by jordan-d-murphy** [comment]
Opened [DOC: Enforce Numpy Docstring Validation (Parent Issue) #58063](https://github.com/pandas-dev/pandas/issues/58063) as a parent issue for fixing docstrings based on the refactoring in code_checks.sh 

Feel free to swing by and help out! 🙂
